streamalert/shared/helpers/aws_api_client.py

```diff
@@ -182,3 +183,71 @@ def download_fileobj(file_handle, bucket, key, region):
         except ClientError:
             LOGGER.error('An error occurred during S3 DownloadFileobj')
             raise
+
+
+class AwsSsm:
+    """Class containing helper functions for aws ssm"""
+
+    @staticmethod
+    def get_parameter(parameter_name, region, with_decryption=True):
+        """gets the parameter_name from SSM Parameter store
+
+        Args:
+            parameter_name (str): The name of the parameter to fetch
+            region (str): AWS region
+            with_decryption (bool): Should decryption be attempted via this call
+
+        Returns:
+            str: The parameter either encrypted or not
+
+        Raises:
+            ClientError
+        """
+        client = boto3.client('ssm', config=default_config(region=region))
```

Comment:
Did you get a chance to verify if the boto3 ssm client has internal side effects upon instantiation?

Reply:
Nope, but like i mentioned on the previous PR. This isn't going to be used alot since once it is used to pull param `x` this will be cached across invocations so it won't call out to the client

Reply:
🆗

Reply:
hmm - thoughts on adding this as a class property that wouldn't ever need to be recreated for the like of the container?

like:

```python
class AwsSsm:
    SSM_CLIENT = None

    @classmethod
    get_parameter(cls, parameter_name, region, with_decryption=True):
        """
        ...
        """
        cls.SSM_CLIENT = cls.SSM_CLIENT or boto3.client('ssm', config=default_config(region=region))
        
        # Now use cls.SSM_CLIENT instead
        ...
```

FWIW: I don't think this is necessary for the `put_parameter` method since that should only be called when a human is involved